packages/flutter/lib/src/painting/borders.dart

```diff
@@ -21,27 +21,6 @@ enum BorderStyle {
   // if you add more, think about how they will lerp
 }
 
-/// The relative position of the stroke on a [BorderSide] in a [Border] or [OutlinedBorder].
-/// When set to [inside], the stroke is drawn completely inside the widget.
-/// For [center] and [outside], a property such as [Container.clipBehavior]
-/// can be used in an outside widget to clip it.
-/// If [Container.decoration] has a border, the container may incorporate
-/// [BorderSide.width] as additional padding:
-/// - [inside] provides padding with full [BorderSide.width].
-/// - [center] provides padding with half [BorderSide.width].
-/// - [outside] provides zero padding, as stroke is drawn entirely outside.
-enum StrokeAlign {
```

Comment:
why is removing this not a breaking change? the code I wrote a few months ago became red w/o any pointers on why or how to fix it. i had to go into the flutter code, see the last changing commit and go through the PR to understand the reason and cause of the breakage. am i missing something?

Reply:
AFAIK it is complicated, there is no easy way to keep that without making a weird class that acts as a shortcut for doubles, even though I tried, it was weird and non-standard, so BorderSide.strokeAlignInside became the default (you can read the discussion in the original PR). The "part 1" and "part 2" commits were like 2 months apart and strokeAlign was still really new when it was modified, but due to how Flutter stable works, they ended up being like 6 months apart.

Reply:
hmm that makes sense. if only dart supported union types natively.

Reply:
Me too, but we can only hope they prioritize that for 2024.